src/targets/registry.ts

```diff
@@ -81,6 +109,7 @@ export class RegistryTarget extends BaseTarget {
     this.github = getGithubClient();
     this.githubRepo = githubRepo;
     this.registryConfig = this.getRegistryConfig();
+    RegistryTarget.instances.add(this);
```

Comment:
This singleton and the constructor side effect seem surprising from the outside. Could you lift this out of the registry target into generic code? Here's a suggestion:

- `BaseTarget` interface class gets something like `supportsBatching(): bool` and `addConfig()`
- The driver asks the target (class) whether it supports batching, and adds configs rather than publishing right away
- At some point defined by the driver -- probably after the last target of that kind -- it issues the `publish()` call

Reply:
This is the first approach I thought about and I decided against it as it complicates code outside of registry too much and without a good reason. It is unclear if any other target needs or can use batching so to restate my goals in the PR description:

1. Do not require any existing config to change
2. Do not require any other target to support batch-mode
3. **Do not add any RegistryTarget-specific code outside of that class**
4. Keep RegistryTarget class usable w/o a batch-helper or factory pattern

This approach would violate items 2 and 3 if the logic is moved to the base class. I'd only move the logic to the base class if at least one or two more targets would need/benefit from batch mode.

Reply:
> It is unclear if any other target needs or can use batching

This is a good point. I spent some time thinking about it and came to these conclusions:

1. The registry target also does not need batching, strictly speaking. The inconsistent releases mentioned in the original issue are actually solved by resuming the failed release using the lockfile, so we already have a solution for that. It's a nice-to-have addition that we only create a single commit. That said, it's extremely sensible to only do a single commit, and run all of this in batch.
2. For the same reason, `upm` and `brew` would benefit from batching. In practice, we don't have a use for this, however.
3. All other targets we repeat require individual uploads or actions, so there's no point in batching. 

When writing my comment originally, I assumed that we want to make this a generic concept: When a target is repeated, its operation is batched. If we want to keep this specific to the registry target, I would rather propose to change the target config to accept an explicit list of specs and update the 3 SDKs where we need that. This gets rid of implicit behavior.

Reply:
>It's a nice-to-have addition that we only create a single commit. That said, it's extremely sensible to only do a single commit, and run all of this in batch.

Mind you, this PR makes a single push but still has multiple commits for clarity and ease of implementation.

>For the same reason, upm and brew would benefit from batching. In practice, we don't have a use for this, however.

I'd be interested to hear more details, especially for `upm` as I cannot understand how it can benefit from batching.

>When writing my comment originally, I assumed that we want to make this a generic concept: When a target is repeated, its operation is batched.

This is how I started and eventually decided that this seems like a special-case for the registry target rather than being a general concept applicable to other targets.